cmake/modules/AddSwift.cmake

```diff
@@ -1592,6 +1598,13 @@ function(add_swift_library name)
          endif()
        endif()
 
+       # Windows doesn't support the -z,defs flag and RemoteMirror needs to build with undefined symbols.
+       if(NOT "${sdk}" STREQUAL "WINDOWS" AND NOT "${name}" STREQUAL "swiftRemoteMirror")
+          # Add back the flags we removed from the call to HandleLLVMOptions 
+          # in SwiftSharedCMakeConfig.cmake.
+          list(APPEND swiftlib_link_flags_all "-Wl,-z,defs")
```

Comment:
I really dislike all of the special cases in this function and additionally how this calls out windows but really you are working with the complement. The technical debt here makes me unhappy.

That being said, I think for now, this is fine.

Reply:
I certainly agree it's far from an ideal approach, but yes, addressing it would take a larger refactor, and I don't think this is much worse than what surrounds it.